Assignments/Assignment5.sql

```diff
@@ -0,0 +1,233 @@
+
+    FROM game_with_outcome g
+        JOIN pregame_odds po on g.game_id = po.game_id
+    GROUP BY g.game_id;
+
+
+DROP TABLE IF EXISTS `rolling_1`;
+CREATE TABLE rolling_1
+    SELECT b.game_id
+         , Hit
+         , atBat
+         , local_date
+         , batter
+    FROM batter_counts b
+        JOIN game g on (b.game_id = g.game_id) # makes the join on matching the game_ids
+    WHERE atBat > 0;
+
+
+#This query takes the table made in the previous query and joins it on itself so I can get two dates to compare to each
+#other so I can calculate the rolling average.
+CREATE INDEX batter_idx ON rolling_1(batter);
+CREATE INDEX date_idx ON rolling_1(local_date)
+
+
+DROP TABLE IF EXISTS `batting_avg_100_day`;
+CREATE TABLE batting_avg_100_day AS
+    SELECT a.batter
+         , a.game_id
+         , a.local_date
+         , ROUND(SUM(b.Hit)/SUM(b.atBat),3) batting_avg
+    FROM rolling_1 a JOIN rolling_1 b ON
+    b.local_date BETWEEN date_add(a.local_date, INTERVAL -100 DAY) AND a.local_date AND a.batter = b.batter
+    GROUP BY a.batter, a.local_date
+
+DROP TABLE IF EXISTS `add_team_roll`;
+CREATE TABLE add_team_roll AS
+    SELECT
+    bg.batter,
+    bg.game_id,
+    bg.team_id,
+    AVG(ba.batting_avg) team_roll_ba
+    FROM batting_avg_100_day ba
+        JOIN battersInGame bg ON bg.batter = ba.batter AND bg.game_id = ba.game_id
+    GROUP BY game_id, bg.team_id;
+
+#This is very similar to the last query but I joined the game table to get the dates so I
+#can group by the year as well as the batter. This makes it an annual batting average.
+DROP TABLE IF EXISTS `batting_avg_annual`;
+CREATE TABLE batting_avg_annual AS
+    SELECT
+           b.game_id
+           , batter
+           , ROUND(SUM(Hit)/SUM(atBat),3) ba_annual
+           , YEAR(local_date) year
+    FROM batter_counts b
+        JOIN game g on (b.game_id = g.game_id) # makes the join on matching the game_ids
+    WHERE atBat > 0
+    GROUP BY batter, year
+    ORDER BY batter, year;
+
+DROP TABLE IF EXISTS `add_team_annual`;
+CREATE TABLE add_team_annual AS
+    SELECT
+    ba.batter,
+    ba.game_id,
+    bg.team_id,
+    AVG(ba.ba_annual) team_annual_ba
+    FROM batting_avg_annual ba
+        JOIN battersInGame bg ON ba.batter = bg.batter AND ba.game_id = bg.game_id
+    GROUP BY game_id, team_id;
+
+
+DROP TABLE IF EXISTS `join_team_ba_game`;
+CREATE TABLE join_team_ba_game AS
+    SELECT
+        g.game_id,
+        #g.local_date,
+        year(g.local_date) AS year,
+        g.home_team_id,
+        g.home_pitcher,
+        g.home_w,
+        g.home_l,
+        g.home_line,
+        g.wins_loss_odds,
+        ata.team_annual_ba,
+        atr.team_roll_ba,
+        p.pitchesThrown,
+        p.innings_pitched,
+        p.pitches_over_DSLP,
+        p.whip,
+        p.season_over_career_era,
+        p.career_so_over_bb,
+        p.season_so_over_bb,
+        p.diff_so_vs_bb,
+        g.home_team_wins
+
+    FROM game_with_odds g
```

Comment:
Maybe I'm missing this but it seems like you're ignoring stats associated with the away team. You may want to add some stats that compare home team performance with away team performance. That being said, I tried some of those and many were garbage. I had a couple, though, that turned out to be helpful at least from an importance standpoint.